ra2ce/graph/networks_utils.py

```diff
@@ -1089,20 +1089,26 @@ def graph_create_unique_ids(graph: nx.Graph, new_id_name: str = "rfid") -> nx.Gr
 
 def add_missing_geoms_graph(graph: nx.Graph, geom_name: str = "geometry") -> nx.Graph:
     # Not all nodes have geometry attributed (some only x and y coordinates) so add a geometry columns
+    def get_node_by_attribute(graph: nx.Graph, node_number: int) -> Union[dict, None]:
+        for node, data in graph.nodes(data=True):
+            if node == node_number:
+                return data
+        return None
```

Comment:
Where is this being used? I would rather just return an empty dictionary to simplify the return value of this function:
```suggestion
    def get_node_by_attribute(graph: nx.Graph, node_number: int) -> dict:
        for node, data in graph.nodes(data=True):
            if node == node_number:
                return data
        return dict()
```